common/src/test/java/com/box/l10n/mojito/okapi/extractor/AssetExtractorTest.java

```diff
@@ -165,6 +165,47 @@ public void genIdForTest() throws UnsupportedAssetFilterTypeException {
                 "Image in text <br id='p1'/>."));
   }
 
+  @Test
+  public void checkDoNotTranslate() throws UnsupportedAssetFilterTypeException {
+    List<AssetExtractorTextUnit> assetExtractorTextUnitsForAsset =
+        assetExtractor.getAssetExtractorTextUnitsForAsset(
+            "test.html",
+            """
+                <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN"
+                    "http://www.w3.org/TR/REC-html40/loose.dtd">
+                <html xmlns="http://www.w3.org/1999/xhtml">
+                <body>
+                <table>
+                  <tr>
+                    <td>
+                      <a title="Visit" href="https://www.somesite.com">
+                      <img src="https://www.somesite.com/img.png">
+                    </a>
+                    </td>
+                    <td>
+                      <img src="https://www.somesite.com/img2.png">
+                    </td>
+                  </tr>
+                </table>
+                </body>
+                </html>
+                """,
+            FilterConfigIdOverride.HTML_ALPHA,
+            Arrays.asList("processImageUrls=false"));
+
+    Assertions.assertThat(assetExtractorTextUnitsForAsset)
+        .extracting(AssetExtractorTextUnit::getName, AssetExtractorTextUnit::getSource)
+        .containsExactly(
+            tuple("5e706a9b7eb9b142a619e4331d6976e6-d41d8cd98f00b204e9800998ecf8427e-1", "Visit"));
+
+    // without filtering according to Okapi "hasText()", it would return translatable text unit
+    // with the following content:
+    // tuple("93e7c3bf7f959f5dbd981361605b4b58-5e706a9b7eb9b142a619e4331d6976e6-1",
+    // "<u id='1'> <br id='p2'/> </u>"),
+    // tuple("e063c00ad4343fedbb80d79301575724-93e7c3bf7f959f5dbd981361605b4b58-1",
+    // "<br id='p1'/>"))
```

Comment:
I don't see "<br id='p1'/>" inside the html content above. Did you forget to add it in this test case by any chance?

Reply:
actually looking closer the other test cases do not have the `br id='p1'` in the html content as well. Not sure from where it comes from. Probably the other test cases in this class which asserts on `br id='p1'` also need to be fixed.

Reply:
HTML: tr/td/img combo was getting converted into a `<u id='1'> <br id='p2'/> </u>` which will end up showing up in the frontend. `<u id='1'> <br id='p2'/> </u>` is not markup that was in the orginal HTML.

In the comment, it is what was returned by the extractor before my fix. I just keep it for the record/explanation. In other words if removing `!textUnit.getSource().hasText()` it would generate 2 extra tuple in that test.